Mathlib/Data/Seq/WSeq.lean

```diff
@@ -1356,12 +1363,15 @@ theorem destruct_ofSeq (s : Seq α) :
 
 @[simp]
 theorem head_ofSeq (s : Seq α) : head (ofSeq s) = Computation.pure s.head := by
-  simp [head]; cases Seq.head s <;> rfl
+  simp only [head, Option.map_eq_map, destruct_ofSeq, Computation.map_pure, Option.map_map]
+  cases Seq.head s <;> rfl
```

Comment:
I'm honestly not sure this kind of changes is an improvement

Reply:
I reverted this change.